src/mlpack/methods/ann/layer/recurrent_impl.hpp

```diff
@@ -126,6 +127,36 @@ Recurrent<InputDataType, OutputDataType, CustomLayers...>::Recurrent(
   this->network.push_back(recurrentModule);
 }
 
+template<typename InputDataType, typename OutputDataType,
+         typename... CustomLayers>
+size_t Recurrent<InputDataType, OutputDataType, CustomLayers...>::InputShape() const
+{
+  size_t inputShapeStartModule = boost::apply_visitor(InShapeVisitor(), startModule);
+  size_t inputShapeInputModule = boost::apply_visitor(InShapeVisitor(), inputModule);
+  size_t inputShapeFeedbackModule = boost::apply_visitor(InShapeVisitor(), feedbackModule);
+  size_t inputShapeTransferModule = boost::apply_visitor(InShapeVisitor(), transferModule);
+
+  if (inputShapeStartModule != 0)
```

Comment:
```suggestion
  // Return the size of the first module that we have.
  if (inputShapeStartModule != 0)
```

It might be nice to add a comment detailing why we have to have logic like this. :+1:

Reply:
Yes, comments will be helpful here, thanks 👍